Makefile.build

```diff
@@ -570,6 +570,11 @@ kernel/kernel.cma: kernel/kernel.mllib
 	$(SHOW)'OCAMLC -a -o $@'
 	$(HIDE)$(OCAMLC) $(MLINCLUDES) $(BYTEFLAGS) $(VMBYTEFLAGS) -a -o $@ $(filter-out %.mllib, $^)
 
+# We need to explicitly list this dependency due to the mixed-handling
```

Comment:
Can someone explain this comment to me?  What is "mixed-handling of modules", and why does it mandate listing this dependency explicitly?

Reply:
Mixed in the sense that interfaces and implementations are managed fully independently, then, it could be the case that the `cmi` is not properly build before the `cmo` and thus parallel compiling will fail in coqchk I believe.

Reply:
If that's the case, why do we need it here but not for all files?

Reply:
This file is a bit special as it is generated automatically and doesn't belong to any cma [by design] that said it could be that it works well without the rule but I was not feeling sure. Note that indeed we have a race in the current build system and I see it fail from time to time due to `cmi` problems.